x-pack/test/security_solution_ftr/page_objects/timeline/index.ts

```diff
@@ -0,0 +1,147 @@
+    await this.pageObjects.header.waitUntilLoadingHasFinished();
+  }
+
+  /**
+   * Ensure that the timeline bottom bar is accessible
+   */
+  async ensureTimelineAccessible(): Promise<void> {
+    await this.testSubjects.existOrFail(TIMELINE_BOTTOM_BAR_CONTAINER_TEST_SUBJ);
+  }
+
+  async findTimelineBottomBarAddButton(): Promise<WebElementWrapper> {
+    await this.ensureTimelineAccessible();
+    return this.testSubjects.findDescendant(
+      'settings-plus-in-circle',
+      await this.testSubjects.find(TIMELINE_BOTTOM_BAR_CONTAINER_TEST_SUBJ)
+    );
+  }
+
+  async showOpenTimelinePopupFromBottomBar(): Promise<void> {
+    await this.ensureTimelineAccessible();
+    await (await this.findTimelineBottomBarAddButton()).click();
+    await this.testSubjects.existOrFail('timeline-addPopupPanel');
+  }
+
+  async openTimelineById(id: string): Promise<void> {
+    await this.showOpenTimelinePopupFromBottomBar();
+    await this.testSubjects.click('open-timeline-button');
+
+    const timelineSelectModel = await this.testSubjects.find('open-timeline-modal');
+
+    await (await this.testSubjects.findDescendant(`title-${id}`, timelineSelectModel)).click();
+    await this.ensureTimelineIsOpen();
+  }
+
+  async closeTimeline(): Promise<void> {
+    if (await this.testSubjects.exists(TIMELINE_CLOSE_BUTTON_TEST_SUBJ)) {
+      await this.testSubjects.click(TIMELINE_CLOSE_BUTTON_TEST_SUBJ);
+      await this.testSubjects.waitForHidden(TIMELINE_MODAL_PAGE_TEST_SUBJ);
+    }
+  }
+
+  async ensureTimelineIsOpen(): Promise<void> {
+    await this.testSubjects.existOrFail(TIMELINE_MODAL_PAGE_TEST_SUBJ);
+  }
+
+  /**
+   * From a visible timeline, clicks the "view details" for an event on the list
+   * @param index
+   */
+  async showEventDetails(index: number = 0) {
+    await this.ensureTimelineIsOpen();
+
+    const event = (await this.testSubjects.findAll('event'))[index];
+
+    if (!event) {
+      throw new Error(`Timeline event at index [${index}] not found`);
+    }
+
+    await (await this.testSubjects.findDescendant('expand-event', event)).click();
+    await this.testSubjects.existOrFail('eventDetails');
+  }
+
+  /**
+   * Clicks the Refresh button at the top of the timeline page and waits for the refresh to complete
+   */
+  async clickRefresh(): Promise<void> {
+    await this.ensureTimelineIsOpen();
+
+    // There are multiple buttons on the page with the same test subject as the Refresh button.
+    // We specifically want the one that is always visible at the top of timeline page.
+    const timelineContentQueryArea = await this.testSubjects.find('timeline-tab-content-query');
+    const refreshButton = await this.testSubjects.findDescendant(
+      'superDatePickerApplyTimeButton',
+      timelineContentQueryArea
+    );
+    await refreshButton.click();
+
+    await this.retry.waitFor(
+      'Timeline refresh button to be enabled',
+      async (): Promise<boolean> => {
```

Comment:
this `async` is not needed since you are not awaiting.

Reply:
The function needs to return a Promise, so I just used `async` in order to avoid returning a `Promise.resolve()`.

Reply:
I see